pkg/build/builder/transient_mounts.go

```diff
@@ -0,0 +1,106 @@
+package builder
+
+import (
+	"fmt"
+	"strings"
+)
+
+// TransientMountOptions are options that change how a transient mount is setup
+type TransientMountOptions struct {
+	NoDev    bool
+	NoExec   bool
+	NoSuid   bool
+	ReadOnly *bool
+}
+
+//TransientMount is a structure to hold the data needed to construct a transient mount
+type TransientMount struct {
+	Source      string
+	Destination string
+	Options     TransientMountOptions
+}
+
+// returns the string representation of a transient mount
+// format: source:destiation:option1,option2,...
+func (tm TransientMount) String() string {
```

Comment:
don't we typically employ `tm *TransientMount)` here @coreydaley ?

if there is a specific reason why you did not do it that way, I'm missing it (and perhaps a clarifying comment wrt that is needed)

Reply:
Since this method is not mutating the underlying data structure the pointer type is not needed. I'm not sure that a comment for that is needed.

https://gobyexample.com/methods

Reply:
I don't think this one should be marked resolved yet @coreydaley

Reply:
updated in current push